app/Actions/Place/GetPlaceCollection/GetPlaceCollectionPresenter.php

```diff
@@ -2,90 +2,72 @@
-                'creator_id' => $photo->creator_id
-            ];
-        }
-
-        return $placeArray;
+    public function __construct(
+        PlacePresenter $placePresenter,
+        ReviewPresenter $reviewPresenter,
+        LocalizationPresenter $localizationPresenter,
+        CityPresenter $cityPresenter,
+        FeaturePresenter $featurePresenter,
+        CategoryPresenter $categoryPresenter,
+        CategoryTagsPresenter $tagsPresenter,
+        PlacePhotoPresenter $photoPresenter
+    ) {
+        $this->placePresenter = $placePresenter;
+        $this->reviewPresenter = $reviewPresenter;
+        $this->localizationPresenter = $localizationPresenter;
+        $this->cityPresenter = $cityPresenter;
+        $this->featurePresenter = $featurePresenter;
+        $this->categoryPresenter = $categoryPresenter;
+        $this->tagsPresenter = $tagsPresenter;
+        $this->photoPresenter = $photoPresenter;
     }
 
-    private static function presentReview(Review $review, ?int $userId): array
+    public function present(GetPlaceCollectionResponse $placeResponse): array
     {
-        $result = [
-            'id' => $review->getKey(),
-            'description' => $review->description,
-            'user' => !is_null($review->user->info) ? [
-                'first_name' => $review->user->info->first_name,
-                'last_name' => $review->user->info->last_name,
-                'avatar_url' => $review->user->info->avatar_url,
-            ] : null,
-            'likes' => $review->likes->count(),
-            'dislikes' => $review->dislikes->count(),
-            'liked' => $review->isLiked($userId),
-            'disliked' => $review->isDisliked($userId)
-        ];
-
-        return $result;
-    }
+        return $placeResponse->getPlaceCollection()->map(function ($place) use ($placeResponse) {
+            $result = $this->placePresenter->present($place);
+            $review = $placeResponse->getReviews()->first(function ($item) use ($place) {
+                return $place->id === $item->place_id;
+            });
+            $result['review'] = $review ?
+                $this->reviewPresenter->present($review, $placeResponse->getUserId()) :
+                null;
+            $result['photos'] = $place->photos->map(function ($photo) {
+                return $this->photoPresenter->present($photo);
+            });
+            $result['city'] = $this->cityPresenter->present($place->city);
+            $result['features'] = $place->features->map(function ($feature) {
+                return $this->featurePresenter->present($feature);
+            });
+            $result['localization'] = $place->localization->map(function ($localization) {
+                return $this->localizationPresenter->present($localization);
+            });
+            $result['category'] = $this->categoryPresenter->present($place->category);
+            $result['category']['tags'] = $place->category->tags->map(function ($tag) {
```

Comment:
I'm not really sure that we will needs this presenter. Why would we need to present category with ALL tags? When we will implement place tags on backend we won't need this relation any more, we will use place->tags instead of place->category->tags, so I probably will leave it like this for now